lib/OpenAPI/Client.pm

```diff
@@ -37,8 +38,26 @@ has ua => sub { Mojo::UserAgent->new };
 sub call {
   my ($self, $op) = (shift, shift);
   my $code = $self->can($op);
-  return $self->$code(@_) if $code;
-  Carp::croak('[OpenAPI::Client] No such operationId');
+  Carp::croak('[OpenAPI::Client] No such operationId') if !$code;
+  $self->$code(@_);
+}
+
+sub call_p {
+  my ($self, $op) = (shift, shift);
+  my $code = $self->can($op);
+  my $promise = Mojo::Promise->new;
+  if (!$code) {
+    return $promise->reject('[OpenAPI::Client] No such operationId');
+  }
+  $self->$code(@_, sub {
+    my ($self, $tx) = @_;
+    my $err = $tx->error;
+    return $promise->reject($err->{message}) if $err && !$err->{code};
+    return $promise->reject('WebSocket handshake failed')
+      if $tx->req->is_handshake && !$tx->is_websocket;
+    $promise->resolve($tx);
+  });
+  $promise;
```

Comment:
How about just using the `call()` method?

```
sub call_p {
  my $self = shift;
  my $promise = Mojo::Promise->new;

  $self->call(@_, sub {
    my ($self, $tx) = @_;
    my $err = $tx->error;
    return $promise->reject($err->{message}) if $err and !$err->{code};
    return $promise->resolve($tx);
  });

  return $promise;
}
```

Reply:
Previously, that was so that I could make a rejected Promise on no such operation. Since you don't like that, I'll do exactly this!